src/transformStateWithClones.js

```diff
@@ -7,7 +7,38 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  const tempState = {
+    ...state,
+  };
+
+  for (const action of actions) {
+    const { type, extraData, keysToRemove } = action;
+
+    switch (type) {
+      case 'addProperties':
+        Object.assign(tempState, extraData);
+        break;
+
+      case 'clear':
+        for (const key in tempState) {
+          delete tempState[key];
+        }
+        break;
+
+      case 'removeProperties':
+        for (const key of keysToRemove) {
+          delete tempState[key];
+        }
+        break;
+
+      default:
+        continue;
```

Comment:
it’s better to add some kind of error, because the cases should have worked for us